src/bit_machine/mod.rs

```diff
@@ -494,13 +494,17 @@ impl BitMachine {
         let output_width = jet.target_ty().to_bit_width();
         // Input buffer is implicitly referenced by input read frame!
         // Same goes for output buffer
-        let (input_read_frame, _input_buffer) = unsafe { get_input_frame(self, input_width) };
+        let (input_read_frame, input_buffer) = unsafe { get_input_frame(self, input_width) };
         let (mut output_write_frame, output_buffer) = unsafe { get_output_frame(output_width) };
 
         let jet_fn = jet.c_jet_ptr();
         let c_env = J::c_jet_env(env);
         let success = jet_fn(&mut output_write_frame, input_read_frame, c_env);
 
+        if cfg!(feature = "trace") {
+            println!("{:?} {:?} -> {:?}", jet, input_buffer, output_buffer);
+        }
```

Comment:
f6312648cd8d3c4f4649ae01639a146992dce8c9: I feel that we shouldn't add print statements to the code, even when they are feature-gated. Printing could be handled via a trait that is implemented in a downstream crate.

Reply:
Oh sorry, that must be in the previous commit, I should squash them